File: dp_labs_and_solutions/dp_labs_and_solutions/dp_matlab_labs_and_solutions/singleton/flyweight/lab/SingleJourney.m
% SingleJourney.m

classdef SingleJourney < Journey
	properties (GetAccess = private, SetAccess = private)
		Address % String
	end

	methods
		% constructor:
		function theSingleJourney = SingleJourney(aAddress)
			if isempty(aAddress)
				error('Invalid aAddress.');
			end

			theSingleJourney.Address = aAddress;
		end

		function output(theSingleJourney)
			fprintf('single journey to %s.\n', theSingleJourney.Address);
		end
	end
end

File: dp_labs_and_solutions/dp_labs_and_solutions/dp_cpp_labs_and_solutions/builder/solution/single_journey.cpp
// single_journey.cpp

#include <iostream>
#include <string>
#include "dp_exception.h"
#include "single_journey.h"

single_journey::single_journey(const std::string & a_address)
{
	if (a_address == "")
	{
		throw dp_exception("single_journey::single_journey(const std::string & a_address) -- a_address == \"\".");
	}
	my_address = a_address;
}

void single_journey::output(std::ostream & a_ostream) const
{
	a_ostream << "single journey to " << my_address << ".\n";
}

int single_journey::get_number_of_stops() const
{
	return 1;
}
